thrust/examples/counting_iterator.cu

```diff
@@ -35,7 +27,7 @@ int main()
   // indices now contains [1,2,5,7]
 
   // print result
-  std::cout << "found " << (indices_end - indices.begin()) << " nonzero values at indices:\n";
+  std::cout << "found " << std::distance(indices.begin(), indices_end) << " nonzero values at indices:\n";
```

Comment:
I believe this should actually use `cuda::std::distance`

Whenever possiblle we should use facilities from `cuda::std` rather than their `std` counterparts.

That will ensure that it also works on device